gloo/cuda_private.cu

```diff
@@ -8,7 +8,10 @@
 
 #include "gloo/cuda_private.h"
 
+#include <array>
```

Comment:
The clang version on macOS complained about this include. It is used further down in this file and wasn't included explicitly. I think older compilers were OK with this because it was included through some other part of the standard library, and that must have changed.